File: src/LeetCode.Easy/Solutions/TwoSumSolution.cs
namespace LeetCode.Easy.Solutions;

public static class TwoSumSolution
{
    public static int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> indexMap = [];
        
        for (var index = 0; index < nums.Length; index++)
        {
            var current = target - nums[index];

            if (indexMap.TryGetValue(current, out var value))
            {
                return [value, index];
            }
            
            indexMap[nums[index]] = index;
        }

        return [];
    }
}

File: Rust/leet_code_easy/src/easy/two_sum_solution.rs
use std::collections::HashMap;

pub fn run() {
    let response = two_sum(vec!(2,7,11,15), 9);
    println!("EAST => TWO SUM => RESPONSE => || {:?} ||", response);
}

fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {

    let mut map: HashMap<i32, i32> = HashMap::new();
    let mut index: i32 = 0;
    for number in nums {

        
        let current = target - number;

        if map.contains_key(&current) {
            return vec![map[&current], index];
        }

        map.insert(number, index);
        index += 1;
    }

    Vec::new()
}
